src/osd/OSD.cc

```diff
@@ -10513,6 +10513,7 @@ void OSD::ShardedOpWQ::_enqueue(OpQueueItem&& item) {
   sdata->shard_lock.lock();
 
   dout(20) << __func__ << " " << item << dendl;
+  bool empty = sdata->pqueue->empty();
```

Comment:
this looks wrong to me.
we might still have to process the on_commit callbacks even if the op_wq is empty.